### ISSUE TITLE
Conditionally include extensions on the command line

### ISSUE THREAD

**Issue by benholmen** [sebastianbergmann/phpunit#5519]
Hello, I've written an extension using the documentation and it works as expected when I enable it in my `phpunit.xml` `<extensions>` config. My extension is essentially a custom printer using the event system.

I would like users to be able to conditionally enable or disable it at run time, as it may not be desirable all of the time. Enabling and disabling via `phpunit.xml` is pretty burdensome, so I would like to be able to do it with a command line option.

I noticed that versions before 10.x included an `--extensions` command line option that is no longer present in 10.x. Is there a replacement?

Alternative: allow extensions to register valid command line options to control their behavior. This would allow me to run my extension, but only do the expected action if the user provided a particular `--option` flag.

Are any of these possible with the current implementation or would they be new features? Thanks for the help! And thanks most of all for PHPUnit!

**Comment by sebastianbergmann** [comment]
The design of the test runner currently assumes that an extension that is configured in the XML configuration file should always be loaded and registered. I did not think about a use case for disabling an extension that is configured in the XML configuration file using a CLI option. This use case could be supported by a `--without-extension className` CLI option where `className` is the name of the extension's bootstrap class that is configured in the XML configuration file.

Currently, you can only use the `--no-extensions` CLI option to disable the loading of all extensions that are configured in `phpunit.xml`.

I do not remember exactly how the `--extensions` CLI option worked in PHPUnit 8.5 and PHPUnit 9.6, but I am almost certain that it did not do what you think it does.

> allow extensions to register valid command line options to control their behavior

In the long run: maybe. But this will not be trivial to get right.

**Comment by benholmen** [comment]
Thanks for the response @sebastianbergmann! Can you think of a way I could allow users to conditionally enable (or disable) my extension on the command line? Maybe I'm not considering something or aware of a feature.

**Comment by nikophil** [comment]
Couldn't you use an environment variable?
```
$ DISABLE_EXTENSION=1 vendor/bin/phpunit
```

**Comment by sebastianbergmann** [comment]
I do not think that adding CLI options such as `--without-extension` or allowing extensions to register their own CLI options warrants the added complexity.

As @nikophil suggested, using an environment variable to control your extension's behavior at runtime sounds good enough to me:

```
DISABLE_EXTENSION=1 vendor/bin/phpunit
```

Your extension's bootstrap class can simply check for this environment variable and act (or not act) accordingly. This approach is simple, does not require any changes to PHPUnit, and gives extension authors full control.
